auth0/src/main/java/com/auth0/android/request/internal/BaseRequest.kt

```diff
@@ -96,12 +96,18 @@ internal open class BaseRequest<T, U : Auth0Exception>(
         val reader = AwareInputStreamReader(response.body, Charset.defaultCharset())
         if (response.isSuccess()) {
             //2. Successful scenario. Response of type T
-            val result: T = resultAdapter.fromJson(reader)
+            val result: T = try {
+                resultAdapter.fromJson(reader)
+            } catch (exception: Exception) {
+                //3. Network exceptions, timeouts, etc reading response body
+                val error: U = errorAdapter.fromException(exception)
+                throw error
```

Comment:
you should `reader.close()` before throwing the error

Reply:
Fixed https://github.com/auth0/Auth0.Android/pull/486/commits/1520d5bf11a74927e17d1de738b59073726d79a2